src/test/java/picard/sam/markduplicates/MarkDuplicatesSetSizeHistogramTester.java

```diff
@@ -0,0 +1,126 @@
+            Assert.assertEquals(outputRecords, this.getNumberOfRecords(), ("saw " + outputRecords + " output records, vs. " + this.getNumberOfRecords() + " input records"));
+
+            // Check the values written to metrics.txt against our input expectations
+            final MetricsFile<DuplicationMetrics, Double> metricsOutput = new MetricsFile<DuplicationMetrics, Double>();
+            try{
+                metricsOutput.read(new FileReader(metricsFile));
+            }
+            catch (final FileNotFoundException ex) {
+                System.err.println("Metrics file not found: " + ex);
+            }
+            Assert.assertEquals(metricsOutput.getMetrics().size(), 1);
+            final DuplicationMetrics observedMetrics = metricsOutput.getMetrics().get(0);
+            Assert.assertEquals(observedMetrics.UNPAIRED_READS_EXAMINED, expectedMetrics.UNPAIRED_READS_EXAMINED, "UNPAIRED_READS_EXAMINED does not match expected");
+            Assert.assertEquals(observedMetrics.READ_PAIRS_EXAMINED, expectedMetrics.READ_PAIRS_EXAMINED, "READ_PAIRS_EXAMINED does not match expected");
+            Assert.assertEquals(observedMetrics.UNMAPPED_READS, expectedMetrics.UNMAPPED_READS, "UNMAPPED_READS does not match expected");
+            Assert.assertEquals(observedMetrics.UNPAIRED_READ_DUPLICATES, expectedMetrics.UNPAIRED_READ_DUPLICATES, "UNPAIRED_READ_DUPLICATES does not match expected");
+            Assert.assertEquals(observedMetrics.READ_PAIR_DUPLICATES, expectedMetrics.READ_PAIR_DUPLICATES, "READ_PAIR_DUPLICATES does not match expected");
+            Assert.assertEquals(observedMetrics.READ_PAIR_OPTICAL_DUPLICATES, expectedMetrics.READ_PAIR_OPTICAL_DUPLICATES, "READ_PAIR_OPTICAL_DUPLICATES does not match expected");
+            Assert.assertEquals(observedMetrics.PERCENT_DUPLICATION, expectedMetrics.PERCENT_DUPLICATION, "PERCENT_DUPLICATION does not match expected");
+            Assert.assertEquals(observedMetrics.ESTIMATED_LIBRARY_SIZE, expectedMetrics.ESTIMATED_LIBRARY_SIZE, "ESTIMATED_LIBRARY_SIZE does not match expected");
+            Assert.assertEquals(observedMetrics.SECONDARY_OR_SUPPLEMENTARY_RDS, expectedMetrics.SECONDARY_OR_SUPPLEMENTARY_RDS, "SECONDARY_OR_SUPPLEMENTARY_RDS does not match expected");
+
+
+            // Check contents of set size bin against expected values //
+            for (final Histogram<Double> histo : metricsOutput.getAllHistograms()) {
+                final String label = histo.getValueLabel();
+                for (Double bin :histo.keySet()) {
+                    String binStr = bin.toString();
+                    if (expectedSetSizeMap.containsKey(Arrays.asList(label,binStr))) {
+                        Histogram.Bin<Double> binValue = histo.get(bin);
+                        final double actual = binValue.getValue();
+                        final double expected = expectedSetSizeMap.get(Arrays.asList(label, binStr));
```

Comment:
the code builds `Arrays.asList(label, binStr)` twice it would be clearer to build it once